diff --git a/types/index.d.ts b/types/index.d.ts
index d23ea1113366ad1d5a3300268ab8c49440dfd0be..0eea14653663ab70c4c62194d4715e0034c888b5 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,6 +1,3 @@
-// This references the old types on purpose, to make sure to not create breaking changes for TS consumers.
-// eslint-disable-next-line @typescript-eslint/triple-slash-reference
-/// <reference path="../../utils/types/index.d.ts"/>
 import * as UppyUtils from '@uppy/utils'
 
 // Utility types
